test/e2e/garbage_collector.go

```diff
@@ -72,6 +106,30 @@ func newOwnerRC(f *framework.Framework, name string) *v1.ReplicationController {
 	}
 }
 
+// verifyRemainingObjects verifies if the number of the remaining deployment
+// and rs are deploymentNum and rsNum. It returns error if the
+// communication with the API server fails.
+func verifyRemainingObjectsAfterDeploymentDeletion(f *framework.Framework, clientSet clientset.Interface, deploymentNum, rsNum int) (bool, error) {
+	rs, err := clientSet.Extensions().ReplicaSets(f.Namespace.Name).List(v1.ListOptions{})
```

Comment:
Can you make sure that the replica sets you are listing are owned by the provided deployment?